### ISSUE TITLE
Basic implementation for iOS >= 9.3

### ISSUE THREAD

**Issue by lukhof** [sc4v3ng3r/flutter_audio_query#43]
Hi,
first thank you for this package! 
I needed iOS support, so I [forked](https://github.com/lukhof/flutter_audio_query/tree/development) your project and did a basic implementation for iOS >= 9.3 in Swift, that I want to contribute.

What's working:
- all read-only methods for all **local** files except for artworks (I'm working on it)
- basic sorting (i still have to implement real alphabetical sorting)

What's not working:
- create/change playlists (never will, these are read-only [see Apple Developer Documentation](https://developer.apple.com/documentation/mediaplayer/mpmediaplaylist) )
- some properties are not supported in iOS, like "year" or "bookmark" for songs

How would you like to handle iOS specific things like:

- cloud items (I filtered them out, because you can't play them)
- Podcasts (they will appear as normal songs, but I can put them in a new category, like playlists)
- not supported properties (I just return "n/a" for the corresponding field in the map)

It's still a work in progress, needs some optimisation, but I will work on it in the next few days.
If everything works well I will create a pull request!

**Comment by sc4v3ng3r** [comment]
All right!!
Thanks @lukhof for your contribution. It will be really appreciated.
I need some motivation to keep going with this plugin and maybe your iOS contribution can be a trigger for that.

### - So about the plugin features for the first iOS implementation would be great if:

1) All read-only methods for local files with the artwork.

2) Playlists access can be just read-only. Don't care about playlist write methods even on Android all of them will be removed soon.

3) iOS not supported properties can empty for now...

### - About iOS specifics

1) In the first implementation lets keep cloud items out.

2) Podcasts can appear as normal songs we can just set **is_podcast** property as **true** in this case.

**Comment by lukhof** [comment]
I'm glad this may be a little motivation booster for you. ;)

The cloud items are not visible, Podcasts get the **is_podcast = true**, playlists where already supported in the first push.
The artwork may be a bit tricky... in iOS you get a **UIImage** object and not a URL/URI. I **need** this feature, so I will think about a workaround.

I have to do some bug fixing, but as I can see in a short test, the iOS version of my project with the new implementation is working like the android version!

I will let you know if I am ready to make a pull request!

**Comment by sc4v3ng3r** [comment]
> The artwork may be a bit tricky... in iOS you get a **UIImage** object and not a URL/URI. I **need** this feature, so I will think about a workaround.

 Check [this](https://pub.dev/documentation/flutter_audio_query/latest/flutter_audio_query/FlutterAudioQuery/getArtwork.html) approach used on Android >= Q to fetch the artworks. In these API levels isn't possible retrieve artwork URL but you can get the image raw bytes.

**Comment by lukhof** [comment]
yes, this approach may work, because i can get the raw image in different formats, thank you!

**Comment by lukhof** [comment]
yep, this is working for me. I have to speed up the process through asynchronous tasks, but I get the artwork in my app!

**Comment by lukhof** [comment]
I pushed the first version with the synchronous way to get the artworks. Tomorrow I try to change it to async.
There is still a bug, the first time you launch the example app it crashes, because I ask for permissions, but do not wait for the result :D

**Comment by lukhof** [comment]
Hey,
I'm still working on some implementations. I did a mistake: you actually CAN modify a playlist, but you MUST know the UUID from the playlist. The UUID is only known at the creation of the playlist. My solution: 
If you create a playlist the UUID, playlist name and persistentID (something else than UUID) is written to the UserDefaults.
 I still need some time, but I'm working on it....

**Comment by xxjonesriderxx** [comment]
@lukhof I'm using your Version of this library and getting following Error: 

> [core] "Error returned from deamon: Domain=com.apple.accounts Code=9 "(null)""
> [iTunesCloud] ACAccountStore 0x28271b170 - Error retrieving iTunesStore accounts err=Error Domain=com.apple.accounts Code=9 "(null)""
> [iTunesCloud] ACAccountStore 0x28271b170 - Error retrieving local store account err=Error Domain=com.apple.accounts Code=9 "(null)""
> [iTunesCloud] [ICUserIdentityStore] Failed to fetch local storea ccount with error: Error Domain=com.apple.accounts Code=9 "(null)""

**Comment by lukhof** [comment]
Hey,
thank you for your feedback. I'm aware of this error and will push an updated in the next few days! Before then you may ignore this error!
